### PR TITLE
MSEARCH-15, MSEARCH-19: Support subjects and keyword search.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/MSEARCH-15 - Instance - support keyword search fields - (title, contributors, identifiers)
https://issues.folio.org/browse/MSEARCH-19 - Instance - support keyword search subjects field

## Changes:
* Make sure `keyword` inventory search includes:
   * title (string)
   * alternativeTitles (alternativeTitles[].alternativeTitle)
   * indexTitle (string)
   * series (array of strings)
   * contributors (contributors[].name
   * identifiers (all) (identifiers[].value) - see note 1 below
* Support multilang search by `subjects` property;
* Add tests.

### CODE REVIEW

File: src/main/resources/model/instance.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
       "type": "object",
       "properties": {
         "value": {
+          "inventorySearchTypes": "title",
```
Comment (pfilippovEPAM) [inline src/main/resources/model/instance.json:46]: why identifiers start related to the title search? 
title search is finished in https://issues.folio.org/browse/MSEARCH-16

File: src/main/resources/model/instance.json [hunk 2]
```diff
@@ -54,14 +55,14 @@
       "type": "object",
       "properties": {
         "name": {
-          "inventorySearchTypes": "contributors",
+          "inventorySearchTypes": ["contributors", "title"],
```
Comment (pfilippovEPAM) [inline src/main/resources/model/instance.json:58]: the same as above, contributors is not a part of title search

File: src/test/java/org/folio/search/service/metadata/LocalSearchFieldProviderTest.java [hunk 3]
```diff
@@ -85,6 +87,18 @@ void shouldAddStarNotationForMultilangField() {
       .containsExactly("title2.sub3.sub5.*");
   }
 
+  @Test
+  void shouldReturnAllExpectedFieldsForInstanceTitleSearch() {
```
Comment (pfilippovEPAM) [inline src/test/java/org/folio/search/service/metadata/LocalSearchFieldProviderTest.java:91]: this test looks like a redundant:
1. It depends of existing model, which can be modified
2. Behavior of inventory search types already tested in other unit tests

File: src/main/resources/model/instance.json (hunk 1 above)
Comment (bohdan-suprun) [reply src/main/resources/model/instance.json]: Thanks, it should be `keyword` instead.

File: src/test/java/org/folio/search/service/metadata/LocalSearchFieldProviderTest.java (hunk 3 above)
Comment (bohdan-suprun) [reply src/test/java/org/folio/search/service/metadata/LocalSearchFieldProviderTest.java]: This test has been removed.

**Review by hjiebsco — APPROVED** [review]

File: src/main/resources/model/instance.json [hunk 4]
```diff
@@ -26,23 +26,24 @@
...
     "series": {
-      "inventorySearchTypes": "title",
+      "inventorySearchTypes": ["title", "keyword"],
       "index": "multilang"
     },
     "identifiers": {
       "type": "object",
       "properties": {
         "value": {
+          "inventorySearchTypes": ["title", "keyword"],
```
Comment (pfilippovEPAM) [inline src/main/resources/model/instance.json:46]: identifiers should be only keyword, without title search group

Comment (bohdan-suprun) [reply src/main/resources/model/instance.json]: thanks, removed.

**Pull request merged**